src/Microsoft.ML.ImageAnalytics/ImagePixelExtractor.cs

```diff
@@ -352,8 +352,14 @@ private ValueGetter<VBuffer<TValue>> GetGetterCore<TValue>(DataViewRow input, in
                         if (src.PixelFormat != PixelFormat.Format32bppArgb && src.PixelFormat != PixelFormat.Format24bppRgb)
                         {
                             var clone = src.Clone(new Rectangle(0, 0, src.Width, src.Height), PixelFormat.Format32bppArgb);
+                            clone.Tag = src.Tag;
                             src.Dispose();
                             src = clone;
+                            using (var ch = Host.Start(nameof(ImagePixelExtractingTransformer)))
+                            {
+                                ch.Warning($"Encountered image {{0}} of unsupported pixel format {{1}} but converting it to {{2}}.",
```

Comment:
>{0} [](start = 64, length = 3)

You're formatting this string using `$`... so that you can convert `{{0}}` to `{0}`, which is an old style format string, which you then use with `src.Tag`. It might be more straightforward to just use the `$` operator only.

Reply:
Tom, didn't you say that we should be using the old style string formatting in errors to reduce number of strings?
Also if you are using the $ style, you should be making it like $"Encountered image {src.Tag } ...

---
In reply to: [280178155](https://github.com/dotnet/machinelearning/pull/3625#discussion_r280178155) [](ancestors = 280178155)

Reply:
> Tom, didn't you say that we should be using the old style string formatting in errors to reduce number of created strings?

I don't think I said that. I have generally been an enthusastic user of `$` style formatting since its introduction, so it would be odd of me to say that. Maybe someone else said it? (Maybe if no one is listening on the pipe, it doesn't bother to format the string? That might be a valid point. But I don't think I'd make that point.)

But even if I bought that, we'd still be having what amounts to the worst of both worlds with this unnecessary `$` which is not a constant. So you start with the `$` style string, which is creating a format string just for the sake of using the old style. It's like I did something like: `string.Format($"{{0}}", a)` instead of just `$"{a}".

> Also if you are using the $ style, you should be making it inline, like so: $"Encountered image {src.Tag } ...

Right, that is correct.